File: Raven.Client.Lightweight/Document/DTC/VolatileOnlyTransactionRecoveryStorage.cs
using System;
using System.Collections.Generic;
using System.IO;

namespace Raven.Client.Document.DTC
{
	public class VolatileOnlyTransactionRecoveryStorage : ITransactionRecoveryStorageContext, ITransactionRecoveryStorage
	{
		public void CreateFile(string name, Action<Stream> createFile)
		{
		}

		public void DeleteFile(string name)
		{
			// noop
		}

		public IEnumerable<string> GetFileNames(string filter)
		{
			yield break;
		}

		public Stream OpenRead(string name)
		{
			throw new NotSupportedException();
		}

		public void Dispose()
		{
			
		}

		public ITransactionRecoveryStorageContext Create()
		{
			return this;
		}
	}
}

File: Raven.Client.Java/src/main/java/net/ravendb/client/document/dtc/VolatileOnlyTransactionRecoveryStorage.java
package net.ravendb.client.document.dtc;

import java.io.InputStream;
import java.util.Collections;
import java.util.List;

import net.ravendb.abstractions.closure.Action1;


public class VolatileOnlyTransactionRecoveryStorage implements ITransactionRecoveryStorageContext, ITransactionRecoveryStorage {

  @Override
  public void createFile(String name, Action1<InputStream> createFile) {
  }

  @Override
  public void deleteFile(String name) {
  }

  @Override
  public List<String> getFileNames(String filter) {
    return Collections.EMPTY_LIST;
  }

  @Override
  public InputStream openRead(String name) {
    throw new IllegalStateException("not supported");
  }

  @Override
  public ITransactionRecoveryStorageContext create() {
    return this;
  }
}
